### DISCUSSION TITLE
How to deal with `rmd` notebooks associated to some sessions?

### DISCUSSION THREAD

**Discussion by ccamara** [WarwickRUG/WarwickRUG.github.io#9] [Q&A]
Some sessions have `.Rmd` files included as resources (i.e. https://github.com/WarwickRUG/WarwickRUG.github.io/blob/main/posts/2022-12-08-web-scraping-with-rselenium/resources/RSelenium.Rmd). While useful, this is problematic for some reasons:

1. When rendering a site, quarto tries to render them as well. This takes extra time (which can be dealt with _freeze) but in some cases stops the execution if an error is found (i.e. this one: https://github.com/WarwickRUG/WarwickRUG.github.io/blob/3ceac268b2c4574f9a881492e4f66c8ea2633b68/posts/2023-03-08-impact-of-covid19-on-energy-consumption/resources/2023-03-08-impact-of-covid19-on-energy-consumption.Rmd#L46-L48 ) 
2. `renv()` detects the libraries used in those notebooks and asks to include them in the lock file (which makes sense in many ways and doesn't in any others -we wouldn't want those files to be executed).

As a temporary measure to address problem 1, I've added `eval = FALSE` to those files, so code is not evaluated, i.e., https://github.com/WarwickRUG/WarwickRUG.github.io/blob/3ceac268b2c4574f9a881492e4f66c8ea2633b68/posts/2023-03-08-impact-of-covid19-on-energy-consumption/resources/2023-03-08-impact-of-covid19-on-energy-consumption.Rmd#L9-L11

This addresses 1, but not 2, and also means that anyone wanting to reproduce the notebook will have to edit it. It also adds manual steps for us to prepare the blog post.

We should decide what to do with them. Some options:

- Get rid of those files from this repo, but link them into a different one
- Rename the `*.rmd` into `*.rmd.txt` so quarto and renv ignore them. This should be accompanied with some instructions on the session reminding users to rename files.
- Include them into a zip file vs within a folder.
- ...?

**Comment by EllaKaye** [comment]
Another option could be to use render targets to specify which (types of) files you want to render, and which you don't. That would seem preferable to me that renaming to `*.rmd.txt` to get quarto to ignore them, though I'm not sure whether it would help with renv.

https://quarto.org/docs/projects/quarto-projects.html#render-targets

**Reply by ccamara** [reply]
This looks great, @EllaKaye ! Definitely worth trying, and way better than the hackysh solutions I proposed. Fingers crossed about renv, although I'm not optimistic at all.

**Reply by ccamara** [reply]
I was quickly looking on `{renv}` documentation but I could not find if exceptions can be raised based on files, but they can be added based on package's exceptions:

```r
renv::settings$ignored.packages("<package>")
```

Source: https://rstudio.github.io/renv/articles/faq.html

**Reply by ccamara** [reply]
@EllaKaye just implemented this on [a9b23f9](https://github.com/WarwickRUG/WarwickRUG.github.io/commit/a9b23f91234d4953b0c56d0f97f85c31dcacd6b2) and, as expected, renv tries to install all dependencies in those files.

**Comment by ccamara** [comment]
A simple solution that would address the rendering and the renv (provided https://github.com/WarwickRUG/WarwickRUG.github.io/discussions/9#discussioncomment-7138312 would not address item 2) would be to host those files externally, either in a companion repo at WRUG or asking presenters to host the files themselves and linking them from WRUG's website.
